### PR TITLE
PD-326: Add Hooks README

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes

Add README to hooks package.

🎟 _Jira ticket:_ [PD-326](https://jira.mongodb.org/browse/PD-326)

## 🛠 Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

~[ ] I have added tests that prove my fix is effective or that my feature works~
- [x] I have added necessary documentation (if appropriate)
~[ ] I have run `yarn changeset` and documented my changes~
* Not running changeset, as this will be bundled with the minor version bump when we release useEscapeKey, and I'm not sure that adding this documentation requires a line on the changelog - but let me know if you feel otherwise

### CODE REVIEW

File: packages/hooks/README.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+);
+```
+
+### Properties
+
+#### target
+
+**Type:** `HTMLElement | null`
+
+HTML element that is subscribed to DOM changes.
```
Comment (DesignerDave) [inline packages/hooks/README.md]: The element isn't subscribed. You subscribe to changes to the element.

File: packages/hooks/README.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+#### options
+
+**Type:** `MutationObserverInit`
+
+Object with information about what DOM changes to subscribe to.
```
Comment (DesignerDave) [inline packages/hooks/README.md]: Maybe link to MDN documentation that goes into more depth on this information?

File: packages/hooks/README.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+Callback function to execute inside of MutationObserver instance.
+
+#### enabled
+
+**Type:** `boolean`
+
+**Default:** `true`
+
+Determines whether or not the hook should run, defaults to true.
```
Comment (DesignerDave) [inline packages/hooks/README.md]: See my other `enabled` comment.

File: packages/hooks/README.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+const Example = () => {
+  [refNode, setRefNode] = useElementNode();
+
+  return <div ref={setRefNode}>I am a ref</div>;
+};
+```
+
+## useEventListener
+
+Hook to subscribe to an eventListener
```
[the remark this answers is not in the document]
Comment (brookescarlett) [reply packages/hooks/README.md]: 👍

File: packages/hooks/README.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+```js
+useEventListener('click', handleClick, { enabled });
+```
+
+### Properties
+
+#### type
+
+**Type:** `Global Event Handler`
```
[the remark this answers is not in the document]
Comment (brookescarlett) [reply packages/hooks/README.md]: 👍

File: packages/hooks/README.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+Parameter to specify options passed to the eventListener
+
+#### optional.enabled
+
+**Type:** `boolean`
+
+**Default:** `true`
+
+Determines whether or not the useEffect hook should run.
```
[the remark this answers is not in the document]
Comment (brookescarlett) [reply packages/hooks/README.md]: 👍

File: packages/hooks/README.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+
+Array to be passed to useEffect hook, such that the hook will only run if the array's values have changed.
+
+#### optional.element
+
+**Type:** `Document | HTMLElement`
+
+**Default:** `document`
+
+Value to be passed as target of event handler, will default to document.
```
[the remark this answers is not in the document]
Comment (brookescarlett) [reply packages/hooks/README.md]: 👍

File: packages/hooks/README.md [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+
+### Properties
+
+#### callback
+
+**Type:** `function`
+
+Callback executed when `EscapeKey` is pressed.
+
+#### optional
```
[the remark this answers is not in the document]
Comment (brookescarlett) [reply packages/hooks/README.md]: we can but the first key is options

File: packages/hooks/README.md (hunk 1 above)
Comment (brookescarlett) [reply packages/hooks/README.md]: Need help wording this :)

File: packages/hooks/README.md (hunk 2 above)
Comment (brookescarlett) [reply packages/hooks/README.md]: 👍

File: packages/hooks/README.md (hunk 3 above)
Comment (brookescarlett) [reply packages/hooks/README.md]: 👍

**Pull request merged**